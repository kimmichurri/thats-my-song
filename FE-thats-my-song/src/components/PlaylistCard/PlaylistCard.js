import React from 'react';
import PropTypes from 'prop-types';

export default function PlaylistCard({title, artist, id, deleteSongFromPlaylist }) {
  return (
    <div className='playlist-song-cards'>
      <h3 className='playlist-songs-title'>{title}</h3>
      <p className='playlist-artists-title'>{artist}</p>
      <button 
        className='remove-song-button' 
        onClick={(e) => deleteSongFromPlaylist(e)} 
        value={id}>Remove</button>
    </div>
  )
}

PlaylistCard.propTypes = {
  title: PropTypes.string,
  artist: PropTypes.string,
  id: PropTypes.number
}

