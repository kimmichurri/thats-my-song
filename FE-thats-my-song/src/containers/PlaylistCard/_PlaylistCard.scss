.playlist-songs-title, .playlist-artists-title {
  font-family: $accent-font;
}

.playlist-songs-title {
  font-size: 2.5rem;
}

.playlist-artists-title {
  font-size: 1.75rem;
}

.playlist-song-cards {
  background: rgba(154, 56, 94, 0.8);
  padding: 20px;
  box-shadow: 0px 0px 70px;
  margin: 15px auto;
  width: 200px;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.remove-song-button {
  color: white;
  background: black;
  font-family: $main-font;
  font-size: 1rem;
  border-radius: 5px;
  border: none;
  padding: 5px;
  &:focus {
    outline: 2px solid #5F6E8E;
  }
  &:hover {
    background-color: #120c16;
    color: #B4A7BC;
    cursor: pointer;
    box-shadow: 1px 1px 40px $dark-hover-box-shadow;
  }
}