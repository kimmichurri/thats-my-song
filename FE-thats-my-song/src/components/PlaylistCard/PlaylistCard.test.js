import React from 'react';
import { shallow } from 'enzyme';
import PlaylistCard from './PlaylistCard';

describe('PlaylistCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<PlaylistCard />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});