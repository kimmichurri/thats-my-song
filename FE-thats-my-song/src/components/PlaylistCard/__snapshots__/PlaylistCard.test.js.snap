// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistCard should match the snapshot 1`] = `
"<div>
  <h3 />
  <p />
  <button value={[undefined]}>
    Remove
  </button>
</div>"
`;
